src/tests.rs

```diff
@@ -283,3 +286,820 @@ fn test_user_defined_reranking_model() {
+        -5.94698749e-02,
+        -1.71230233e-03,
+        -4.03233198e-03,
+        1.48299001e-02,
+        7.86300674e-02,
+        1.50310015e-03,
+        -1.16257591e-03,
+        7.00143799e-02,
+        1.91469155e-02,
+        5.80814220e-02,
+        -1.99674275e-02,
+        4.18508872e-02,
+        -5.75775281e-02,
+        2.74508097e-03,
+        6.60069799e-03,
+        1.76840704e-02,
+        3.36040333e-02,
+        -5.98163791e-02,
+        8.07598233e-03,
+        -9.57235135e-03,
+        2.19363044e-03,
+        -3.61714065e-02,
+        -4.86176834e-02,
+        2.04166751e-02,
+        -3.27581167e-02,
+        6.22947924e-02,
+        1.23972511e-02,
+        7.61830062e-02,
+        1.91696007e-02,
+        -4.73358668e-02,
+        5.42867519e-02,
+        2.46673934e-02,
+        -1.99356023e-02,
+        -4.07738797e-02,
+        7.28403917e-03,
+        -5.33810966e-02,
+        6.59581944e-02,
+        -6.88298568e-02,
+        3.51754688e-02,
+        2.11542603e-02,
+        -1.21644046e-02,
+        1.03266397e-02,
+        -4.24144715e-02,
+        -9.27274749e-02,
+        3.97489183e-02,
+        2.34165089e-03,
+        5.20037301e-02,
+        -6.57979399e-02,
+        -8.52417201e-02,
+        -4.14646305e-02,
+        4.82670628e-02,
+        2.74149645e-02,
+        -2.20585123e-01,
+        -2.22670455e-02,
+        8.79362109e-04,
+        2.23514624e-02,
+        1.99962524e-03,
+        2.01164614e-02,
+        -2.74167899e-02,
+        3.45750041e-02,
+        -8.87837168e-03,
+        8.74386821e-03,
+        -4.71623568e-03,
+        3.49107198e-02,
+        5.02648056e-02,
+        -3.27357538e-02,
+        5.25373034e-02,
+        2.83128750e-02,
+        8.64231866e-03,
+        1.19051291e-02,
+        6.29680883e-03,
+        2.74104590e-04,
+        4.32253405e-02,
+        7.08232052e-04,
+        1.02054246e-01,
+        3.39808711e-03,
+        -3.83229554e-02,
+        2.88998429e-03,
+        -7.87280276e-02,
+        -4.61945683e-02,
+        -2.36111972e-02,
+        1.82243399e-02,
+        9.28411484e-02,
+        -3.89853236e-03,
+        8.95828530e-02,
+        -6.15564100e-02,
+        2.39589512e-02,
+        -1.67697333e-02,
+        2.61243731e-02,
+        -5.60313016e-02,
+        -5.83290868e-03,
+        -7.25452080e-02,
+        -4.43094261e-02,
+        2.57209633e-02,
+        -1.94783248e-02,
+        -1.71142481e-02,
+        2.29236241e-02,
+        -6.96708038e-02,
+        -1.15762523e-03,
+        -6.09223694e-02,
+        1.82569548e-02,
+        -6.81673689e-03,
+        -7.86587968e-03,
+        5.43668400e-03,
+        -2.06808243e-02,
+        1.39759853e-02,
+        -3.07560433e-02,
+        1.45595375e-04,
+        -1.08737415e-02,
+        3.26007046e-02,
+        -3.86322886e-02,
+        -1.04548130e-02,
+        4.71166559e-02,
+        1.69442664e-03,
+        -9.29024518e-02,
+        8.17656144e-02,
+        5.08985445e-02,
+    ];
```

Comment:
How about we check only the first 5 elements?
Seems unnecessary to type these many literals here.